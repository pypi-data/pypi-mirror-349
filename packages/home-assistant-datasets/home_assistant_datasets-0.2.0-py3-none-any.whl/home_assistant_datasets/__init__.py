"""
.. include:: ../README.md
"""

__all__ = [
    "tool",
    "datasets",
    "models",
    "agent",
    "entity_state",
    "model_client",
    "scrape",
    "secrets",
    "plugins",
]
