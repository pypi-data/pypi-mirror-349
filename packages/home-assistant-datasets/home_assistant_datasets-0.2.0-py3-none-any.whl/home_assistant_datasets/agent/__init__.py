"""Conversation agent module.

This module is a thin wrapper around the Conversation Agent process service,
tied to a simple interface to handle things like rate limiting, timing,
and retries.

This module is primarily used during data collection / scraping.
"""

from . import rate_limit, retryable, service_call, timing
from .agent import ConversationAgent


__all__ = [
    "ConversatioAgent",
    "create_default_agent",
    "trace_events",
    "service_call",
    "rate_limit",
    "retryable",
    "timing",
]


def create_default_agent(
    conversation_agent_id: str, rpm: int | None = None
) -> ConversationAgent:
    """Create the conversation agent client id.

    Note that this should be long lived in order for rate limiting to be effective.
    """
    agent = service_call.create_agent(conversation_agent_id)
    agent = timing.timed_agent(agent)
    if rpm:
        agent = rate_limit.wrap_rate_limit(agent, rpm)
    agent = retryable.wrap_retryable(agent)
    return agent
