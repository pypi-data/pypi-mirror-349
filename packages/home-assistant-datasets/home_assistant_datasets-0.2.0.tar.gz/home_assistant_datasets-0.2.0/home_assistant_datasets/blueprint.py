"""Dataclasses for blueprints.

This module contains dataclasses for blueprints and extracting blueprint content
from the model output.
"""

import re
from contextlib import contextmanager
from dataclasses import dataclass
import tempfile
from typing import Generator

import yaml


# Regular expression to extract yaml/blueprint from the model output.
YAML_RESPONSE = re.compile(r".*```yaml\s*(.*?)\s+```.*", re.DOTALL | re.MULTILINE)


@dataclass(kw_only=True)
class BlueprintContentStatus:
    """Dataclass to hold the blueprint content generated by the model."""

    valid: bool
    """Whether we could extract valid yaml from the model output."""

    error_details: str | None = None
    """Details about the error extracting the yaml."""

    def assert_valid(self) -> None:
        """Assert that the blueprint content is valid."""
        assert self.valid, self.error_details


VALID_BLUEPRINT = BlueprintContentStatus(valid=True)


@dataclass
class BlueprintContent:
    """Dataclass to hold the blueprint content generated by the model."""

    status: BlueprintContentStatus
    """Status of the blueprint content."""

    filename: str | None
    """Filename of the extracted yaml."""

    yaml_content: str | None
    """YAML content extracted from the model output."""

    def validate_inputs_present(self, expected_inputs: list[str]) -> None:
        """Assert that the specified inputs of the blueprint are present."""
        if self.yaml_content is None:
            raise AssertionError("Blueprint content is not available")
        try:
            doc = yaml.load(
                self.yaml_content, Loader=yaml.BaseLoader
            )  # Ignore constructors
        except yaml.YAMLError as err:
            err_str = str(err).replace("\n", " ")
            assert False, f"Blueprint was not valid yaml: {err_str}"

        blueprint = doc.get("blueprint")
        assert blueprint, "Blueprint yaml did not contain 'blueprint:' key"
        input = blueprint.get("input")
        assert input
        for key in expected_inputs:
            assert key in input, f"Required input '{key}' missing from blueprint"


@contextmanager
def extract_blueprint_content(response: str) -> Generator[BlueprintContent, None, None]:
    """Extract the blueprint content from the model output into a temporary file."""

    m = YAML_RESPONSE.match(response)
    if not m or (match_text := m.group(1)) is None:
        text = response[:100].replace("\n", " ")
        if len(response) > 100:
            text += "..."
        yield BlueprintContent(
            status=BlueprintContentStatus(
                valid=False,
                error_details=f"Could not extract YAML from model response: {text}",
            ),
            filename=None,
            yaml_content=None,
        )
        return

    with tempfile.NamedTemporaryFile("w", suffix=".yaml", delete_on_close=False) as tf:
        filename = tf.name
        tf.write(match_text)
        tf.close()
        yield BlueprintContent(
            status=VALID_BLUEPRINT, filename=filename, yaml_content=match_text
        )
