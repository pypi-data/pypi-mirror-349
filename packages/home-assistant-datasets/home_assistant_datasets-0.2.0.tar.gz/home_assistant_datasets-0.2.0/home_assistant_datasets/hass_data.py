"""A home assistant client library wrapper.

This is deprecated and can be replaced by the synthetic home inventory
collection tools.
"""

from dataclasses import dataclass
import hass_client
from hass_client.models import Area, Device, Entity, State

URL_FORMAT = "wss://{hostport}/api/websocket"


async def create_client(hostport: str, api_token: str) -> hass_client.HomeAssistantClient:
    """Create a new home assistant client."""
    client = hass_client.HomeAssistantClient(
        URL_FORMAT.format(hostport=hostport), api_token
    )
    await client.connect()
    if not client.connected:
        raise ValueError("Failed to connect to home assistant")
    return client


@dataclass
class Home:
    """A home assistant data wrapper."""
    areas: list[Area]
    devices: list[Device]
    entities: list[Entity]
    states: list[State]

    def devices_by_area(self, area_id: str) -> list[Device]:
        """Get all devices in an area."""
        return [device for device in self.devices if device['area_id'] == area_id]

    def entities_by_device_id(self, device_id: str) -> list[Entity]:
        """Get all entities for a device."""
        entities = []
        for entity in self.entities:
            if entity['device_id'] != device_id:
                continue
            if entity['disabled_by'] is not None or entity['hidden_by'] is not None:
                continue
            if entity['entity_category'] is not None:
                continue
            entities.append(entity)
        return entities

    def entities_by_area(self, area_id: str) -> list[Entity]:
        return [
            entity
            for device in self.devices_by_area(area_id)
            for entity in self.entities_by_device_id(device['id'])
        ]

    def get_state(self, entity_id: str) -> State | None:
        """Get the state for an entity."""
        for state in self.states:
            if state['entity_id'] == entity_id:
                return state
        return None


async def get_home(client: hass_client.HomeAssistantClient) -> Home:
    """Get the home assistant data."""
    devices: list[Device] = await client.get_device_registry()
    entities: list[Entity] = await client.get_entity_registry()
    states: list[State] = await client.get_states()
    areas: list[Area] = await client.get_area_registry()
    return Home(areas, devices, entities, states)
