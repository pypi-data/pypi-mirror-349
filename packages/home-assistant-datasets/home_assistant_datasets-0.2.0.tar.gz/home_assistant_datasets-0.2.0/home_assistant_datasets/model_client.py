"""A client library wrapper for simple model completions."""

from openai import OpenAI
import google.generativeai as genai


class OpenAIClient:
    """A client for performing completions on a model."""

    def __init__(self, client: OpenAI, model_id: str):
        """Create a new model client."""
        self.client = client
        self.model_id = model_id

    def complete(self, prompt: str, user_message: str) -> str:
        """Complete a user message."""
        response = self.client.chat.completions.create(
            model=self.model_id,
            messages=[
                {"content": prompt, "role": "system"},
                {"content": user_message, "role": "user"},
            ],
        )
        return "".join([choice.message.content or "" for choice in response.choices])


class GoogleClient:
    """A client for performing completions on a model."""

    def __init__(self, model_id: str):
        """Create a new model client."""
        self.client = genai.GenerativeModel(model_id)

    def complete(self, prompt: str, user_message: str) -> str:
        """Complete a user message."""
        response = self.client.generate_content(f"{prompt}\n{user_message}")
        return response.text or ""
