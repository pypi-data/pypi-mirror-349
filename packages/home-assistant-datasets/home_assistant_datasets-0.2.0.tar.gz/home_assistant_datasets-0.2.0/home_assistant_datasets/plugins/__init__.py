"""pytest plugins for driving evaluation tasks.

There are two primary plugins used by tests:

- pytest_scrape: This module is used when scraping outputs from a model. It is
    configured with a dataset and some models and stores the output in a model
    output directory.
- pytest_metrics: This module is for scoring or evaluating the outputs against
    the specified eval task. This reads the previously scraped output, evaluates
    the model on the tasks, and generates a summary report.

These modules internally use pytest plugins and libraries to perform the entire
workflow.
"""

__all__ = [
    "pytest_synthetic_home",
    "pytest_scrape",
    "pytest_metrics",
]
