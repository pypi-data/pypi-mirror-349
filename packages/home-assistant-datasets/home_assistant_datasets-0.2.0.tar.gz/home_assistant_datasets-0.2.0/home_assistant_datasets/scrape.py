"""Libraries related to scraping model outputs and reading scraped outputs.

The scrape config is also used to record the details of how the scrape
was performed when recording model outputs.
"""

from dataclasses import dataclass, field
import datetime
import getpass
from importlib.metadata import version
import logging
import pathlib
import sys
from typing import Any
import uuid


from mashumaro.mixins.yaml import DataClassYAMLMixin
from mashumaro.config import BaseConfig
import yaml

from .yaml_loaders import yaml_decode

__all__ = [
    "ScrapeConfig",
    "ScrapeContext",
    "read_scrape_context",
    "ModelOutput",
    "ModelOutputWriter",
]

_LOGGER = logging.getLogger(__name__)


SCRAPE_CONTEXT_FILE = "_scrape_context.yaml"

# TODO: Combine the scrape config and model output writing so that the entire
# contents of model scrapes are written together. Rename files model output
# and eval task to be specific to scrapes.

@dataclass(kw_only=True)
class ScrapeConfig(DataClassYAMLMixin):
    """Details about a collection of model outputs collected by this tooling."""

    dataset: str
    """The dataset identifier used to generate the predictions."""

    dataset_path: str
    """The path to the dataset used to generate the predictions."""

    dataset_version: str | None = None
    """Additional version information about the dataset."""

    model_id: str
    """The model identifier used to generate the predictions."""

    model_output_path: str
    """Path where model predictions are stored for this scrape."""

    @property
    def scrape_output_path(self) -> pathlib.Path:
        """Base path for output files."""
        return pathlib.Path(self.model_output_path) / self.model_id

    def eval_task_output_path(self, task_id: str) -> pathlib.Path:
        """Return the output filename for an evaluation task run.

        This output file needs to be unique across the test instances to avoid overwriting. For
        example if you add a parameter based on the system prompt then this needs to create
        a separate file containing an id of the prompt.
        """
        return self.scrape_output_path / f"{task_id}.yaml"


@dataclass(kw_only=True)
class ScrapeContext(DataClassYAMLMixin):
    """Details about a collection of model outputs collected by this tooling."""

    uuid: str
    """Unique identifier of the scrape run."""

    timestamp: datetime.datetime = field(default_factory=datetime.datetime.now)
    """The time when the scrape was performed."""

    scrape_config: ScrapeConfig
    """Details about the scrape configuration."""

    version: str
    """The home assistant software version used to create the scrape record."""

    context: dict[str, str] = field(default_factory=dict)
    """Additional context about this scrape record provided by tooling."""

    notes: str = ""
    """Additional manually entered from the developer about this scrape record."""


def read_scrape_context(filename: pathlib.Path) -> ScrapeContext:
    """Read the scrape context from a file."""
    return yaml_decode(filename.open(), ScrapeContext)


def create_scrape_context(scrape_config: ScrapeConfig) -> ScrapeContext:
    """Fixture to generate a scrape record."""
    home_assistant_version = version("homeassistant")
    context: dict[str, Any] = {"user": getpass.getuser() or "unknown"}
    if sys.argv:
        context["argv"] = sys.argv
    return ScrapeContext(
        uuid=str(uuid.uuid4()),
        timestamp=datetime.datetime.now(),
        version=home_assistant_version,
        scrape_config=scrape_config,
        context=context,
        notes="",
    )


def write_scrape_context(scrape_config: ScrapeConfig) -> None:
    """Write scrape context to the configured output path."""
    scrape_context = create_scrape_context(scrape_config)
    scrape_config.scrape_output_path.parent.mkdir(exist_ok=True)
    scrape_config.scrape_output_path.mkdir(exist_ok=True)
    output = scrape_config.scrape_output_path / SCRAPE_CONTEXT_FILE
    output.write_text(yaml.dump(scrape_context, sort_keys=False, explicit_start=True))


@dataclass(kw_only=True)
class ModelOutput(DataClassYAMLMixin):
    """The prediction result of a model."""

    uuid: str
    """A unique id assigned to this prediction."""

    task_id: str
    """An identifier for the task in the dataset."""

    model_id: str | None = None
    """The model identifier used to generate the prediction."""

    category: str | list[str]
    """A label used for slicing model predictions."""

    task: dict[str, Any]
    """Details about the original task for easier inspection of model predictions."""

    response: str
    """The model prediction."""

    context: dict[str, Any]
    """Additional context about the prediction run, e.g. internal model call details."""

    @property
    def categories(self) -> list[str]:
        """Labels used for slicing model predictions."""
        if isinstance(self.category, list):
            return self.category
        return [self.category]

    class Config(BaseConfig):
        omit_none = True
        sort_keys = False


class ModelOutputWriter:
    """Writes records to the eval output."""

    def __init__(self, filename: pathlib.Path):
        """Initialize ModelOutputWriter."""
        self._eval_output = filename

    def write(self, record: ModelOutput) -> None:
        """Write an eval record."""
        self._eval_output.write_text(
            yaml.dump(
                record.to_dict(),
                sort_keys=False,
                explicit_start=True,
                allow_unicode=True,
            )
        )
