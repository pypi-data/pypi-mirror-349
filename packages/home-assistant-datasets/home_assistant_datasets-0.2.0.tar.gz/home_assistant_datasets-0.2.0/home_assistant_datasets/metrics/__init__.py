"""Module for computing eval related metrics and reporting."""

from typing import Any
from dataclasses import field, dataclass

from home_assistant_datasets.agent.trace_events import TokenStats


__all__ = [
    "ScrapeRecord",
    "TaskResult",
    "GOOD_LABEL",
    "BAD_LABEL",
    "report",
    "report_suite",
    "accuracy",
    "token_stats",
    "report_csv",
]


GOOD_LABEL = "Good"
BAD_LABEL = "Bad"


@dataclass
class ScrapeRecord:
    """Identifier for a particular model scrape/collection task."""

    uuid: str
    """The unique identifier for the model output/prediction."""

    task_id: str
    """An identifier for the task in the dataset."""

    model_id: str
    """The model identifier used to generate the prediction."""

    context: dict[str, Any] = field(default_factory=dict)
    """Additional context/detail from runtime of evaluating the prediction."""

    token_stats: TokenStats | None = None
    """Token statistics for the model output."""

    extra_data: dict[str, Any] | None = None
    """Additional task specific data to include in the detailed eval report."""

    def as_report_data(self) -> dict[str, Any]:
        """Export the record for reporting."""
        return {
            "task_id": self.task_id,
            "model_id": self.model_id,
            **(self.extra_data if self.extra_data is not None else {}),
        }


@dataclass
class TaskResult:
    """Result of a specific measurement task for a scrape record."""

    task_name: str
    """The name of the particular test for this test e.g. "accuracy"."""

    passed: bool
    """If true, the test passed. If false, it failed."""

    detail: str | None = None
    """Additional detail about the test failure."""

    skipped: bool = False
    """Entirely ignore this pass/fail result.."""

    def as_report_data(self) -> dict[str, Any] | None:
        """Export the record for reporting."""
        if self.skipped:
            return None
        return {
            "task_name": self.task_name,
            "label": GOOD_LABEL if self.passed else BAD_LABEL,
            "details": self.detail,
        }
