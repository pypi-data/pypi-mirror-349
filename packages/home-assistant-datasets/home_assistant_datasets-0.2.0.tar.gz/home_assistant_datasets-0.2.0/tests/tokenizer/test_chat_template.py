"""Tests for chat template builders."""

from syrupy import SnapshotAssertion

from home_assistant_datasets.tokenizer import conversation, chat_template


TOOLS = [
    conversation.Tool(
        name="get_current_conditions",
        description="Get the current weather conditions for a specific location",
        parameters={"location": {"type": "string"}},
    )
]
TOOL_CALLS = [
    conversation.ToolCall(
        name="get_current_conditions",
        arguments={"location": "San Francisco, CA"},
    )
]


def test_system(snapshot: SnapshotAssertion) -> None:
    """Test a single system message."""

    prompt = chat_template.build_prompt(
        messages=[
            conversation.Message(role="system", content="You are a helpful assistant.")
        ],
        add_generation_prompt=True,
    )
    assert prompt == snapshot


def test_conversation(snapshot: SnapshotAssertion) -> None:
    """Test a conversation."""

    prompt = chat_template.build_prompt(
        messages=[
            conversation.Message(
                role="system",
                content="You are a helpful assistant.",
            ),
            conversation.Message(role="user", content="What is the capital of France?"),
            conversation.Message(
                role="assistant", content="The capital of France is Paris."
            ),
        ]
    )
    assert prompt == snapshot


def test_system_tools(snapshot: SnapshotAssertion) -> None:
    """Test a system message with tools."""

    prompt = chat_template.build_prompt(
        messages=[
            conversation.Message(
                role="system",
                content="You are a helpful assistant.",
            ),
            conversation.Message(
                role="user",
                content="What is the weather like today in SF?",
            ),
        ],
        tools=TOOLS,
        add_generation_prompt=True,
    )
    assert prompt == snapshot


def test_tool_call(snapshot: SnapshotAssertion) -> None:
    """Test a tool call with output."""

    prompt = chat_template.build_prompt(
        messages=[
            conversation.Message(
                role="system",
                content="You are a helpful assistant.",
            ),
            conversation.Message(
                role="user",
                content="What is the weather like today in SF?",
            ),
            conversation.Message(
                role="assistant",
                tool_calls=TOOL_CALLS,
            ),
            conversation.Message(
                role="tool",
                content={
                    "output": "Clouds giving way to sun Hi: 76째 Tonight: Mainly clear early, then areas of low clouds forming Lo: 56째"
                },
            ),
        ],
        tools=TOOLS,
        add_generation_prompt=True,
    )
    assert prompt == snapshot


def test_tool_call_conversation(snapshot: SnapshotAssertion) -> None:
    """Test a tool call conversation."""

    prompt = chat_template.build_prompt(
        messages=[
            conversation.Message(
                role="system",
                content="You are a helpful assistant.",
            ),
            conversation.Message(
                role="user",
                content="What is the weather like today in SF?",
            ),
            conversation.Message(
                role="assistant",
                tool_calls=TOOL_CALLS,
            ),
            conversation.Message(
                role="tool",
                content={
                    "output": "Clouds giving way to sun Hi: 76째 Tonight: Mainly clear early, then areas of low clouds forming Lo: 56째"
                },
            ),
            conversation.Message(
                role="assistant",
                content="The weather in SF looks cloudy today.",
            ),
        ],
        tools=TOOLS,
    )
    assert prompt == snapshot


def test_converation_record(snapshot: SnapshotAssertion) -> None:
    """Test a training record."""

    record = conversation.ConversationRecord(
        instructions="You are a helpful assistant.",
        input="What is the weather like today in SF?",
        output="I will call a weather tool",
        tools=TOOLS,
        tool_calls=TOOL_CALLS,
    )
    prompt = chat_template.build_prompt(record.to_messages(), record.tools)
    assert prompt == snapshot


def test_converation_record_output(snapshot: SnapshotAssertion) -> None:
    """Test a training record."""

    record = conversation.ConversationRecord(
        instructions="You are a helpful assistant.",
        input="What is the weather like today in SF?",
        output="I can't help you with that.",
        tools=TOOLS,
    )
    prompt = chat_template.build_prompt(record.to_messages(), record.tools)
    assert prompt == snapshot
