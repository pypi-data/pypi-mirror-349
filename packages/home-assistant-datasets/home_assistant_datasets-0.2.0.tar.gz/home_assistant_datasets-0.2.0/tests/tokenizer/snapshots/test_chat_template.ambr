# serializer version: 1
# name: test_converation_record
  '''
  <|begin_of_text|><|start_header_id|>system<|end_header_id|>
  
  Environment: ipython
  Cutting Knowledge Date: December 2023
  Today Date: 26 Jul 2024
  
  You are a helpful assistant.<|eot_id|><|start_header_id|>user<|end_header_id|>
  
  Given the following functions, please respond with a JSON for a function call with its proper arguments that best answers the given prompt.
  
  Respond in the format {"name": function name, "parameters": dictionary of argument name and its value}.Do not use variables.
  
  {
      "type": "function",
      "function": {
          "name": "get_current_conditions",
          "description": "Get the current weather conditions for a specific location",
          "parameters": {
              "location": {
                  "type": "string"
              }
          }
      }
  }
  
  What is the weather like today in SF?<|eot_id|><|start_header_id|>assistant<|end_header_id|>
  
  {"name": "get_current_conditions", "parameters": {"location": "San Francisco, CA"}}<|eot_id|>
  '''
# ---
# name: test_converation_record_output
  '''
  <|begin_of_text|><|start_header_id|>system<|end_header_id|>
  
  Environment: ipython
  Cutting Knowledge Date: December 2023
  Today Date: 26 Jul 2024
  
  You are a helpful assistant.<|eot_id|><|start_header_id|>user<|end_header_id|>
  
  Given the following functions, please respond with a JSON for a function call with its proper arguments that best answers the given prompt.
  
  Respond in the format {"name": function name, "parameters": dictionary of argument name and its value}.Do not use variables.
  
  {
      "type": "function",
      "function": {
          "name": "get_current_conditions",
          "description": "Get the current weather conditions for a specific location",
          "parameters": {
              "location": {
                  "type": "string"
              }
          }
      }
  }
  
  What is the weather like today in SF?<|eot_id|><|start_header_id|>assistant<|end_header_id|>
  
  I can't help you with that.<|eot_id|>
  '''
# ---
# name: test_conversation
  '''
  <|begin_of_text|><|start_header_id|>system<|end_header_id|>
  
  Cutting Knowledge Date: December 2023
  Today Date: 26 Jul 2024
  
  You are a helpful assistant.<|eot_id|><|start_header_id|>user<|end_header_id|>
  
  What is the capital of France?<|eot_id|><|start_header_id|>assistant<|end_header_id|>
  
  The capital of France is Paris.<|eot_id|>
  '''
# ---
# name: test_system
  '''
  <|begin_of_text|><|start_header_id|>system<|end_header_id|>
  
  Cutting Knowledge Date: December 2023
  Today Date: 26 Jul 2024
  
  You are a helpful assistant.<|eot_id|><|start_header_id|>assistant<|end_header_id|>
  
  
  '''
# ---
# name: test_system_tools
  '''
  <|begin_of_text|><|start_header_id|>system<|end_header_id|>
  
  Environment: ipython
  Cutting Knowledge Date: December 2023
  Today Date: 26 Jul 2024
  
  You are a helpful assistant.<|eot_id|><|start_header_id|>user<|end_header_id|>
  
  Given the following functions, please respond with a JSON for a function call with its proper arguments that best answers the given prompt.
  
  Respond in the format {"name": function name, "parameters": dictionary of argument name and its value}.Do not use variables.
  
  {
      "type": "function",
      "function": {
          "name": "get_current_conditions",
          "description": "Get the current weather conditions for a specific location",
          "parameters": {
              "location": {
                  "type": "string"
              }
          }
      }
  }
  
  What is the weather like today in SF?<|eot_id|><|start_header_id|>assistant<|end_header_id|>
  
  
  '''
# ---
# name: test_tool_call
  '''
  <|begin_of_text|><|start_header_id|>system<|end_header_id|>
  
  Environment: ipython
  Cutting Knowledge Date: December 2023
  Today Date: 26 Jul 2024
  
  You are a helpful assistant.<|eot_id|><|start_header_id|>user<|end_header_id|>
  
  Given the following functions, please respond with a JSON for a function call with its proper arguments that best answers the given prompt.
  
  Respond in the format {"name": function name, "parameters": dictionary of argument name and its value}.Do not use variables.
  
  {
      "type": "function",
      "function": {
          "name": "get_current_conditions",
          "description": "Get the current weather conditions for a specific location",
          "parameters": {
              "location": {
                  "type": "string"
              }
          }
      }
  }
  
  What is the weather like today in SF?<|eot_id|><|start_header_id|>assistant<|end_header_id|>
  
  {"name": "get_current_conditions", "parameters": {"location": "San Francisco, CA"}}<|eot_id|><|start_header_id|>ipython<|end_header_id|>
  
  {"output": "Clouds giving way to sun Hi: 76째 Tonight: Mainly clear early, then areas of low clouds forming Lo: 56째"}<|eot_id|><|start_header_id|>assistant<|end_header_id|>
  
  
  '''
# ---
# name: test_tool_call_conversation
  '''
  <|begin_of_text|><|start_header_id|>system<|end_header_id|>
  
  Environment: ipython
  Cutting Knowledge Date: December 2023
  Today Date: 26 Jul 2024
  
  You are a helpful assistant.<|eot_id|><|start_header_id|>user<|end_header_id|>
  
  Given the following functions, please respond with a JSON for a function call with its proper arguments that best answers the given prompt.
  
  Respond in the format {"name": function name, "parameters": dictionary of argument name and its value}.Do not use variables.
  
  {
      "type": "function",
      "function": {
          "name": "get_current_conditions",
          "description": "Get the current weather conditions for a specific location",
          "parameters": {
              "location": {
                  "type": "string"
              }
          }
      }
  }
  
  What is the weather like today in SF?<|eot_id|><|start_header_id|>assistant<|end_header_id|>
  
  {"name": "get_current_conditions", "parameters": {"location": "San Francisco, CA"}}<|eot_id|><|start_header_id|>ipython<|end_header_id|>
  
  {"output": "Clouds giving way to sun Hi: 76째 Tonight: Mainly clear early, then areas of low clouds forming Lo: 56째"}<|eot_id|><|start_header_id|>assistant<|end_header_id|>
  
  The weather in SF looks cloudy today.<|eot_id|>
  '''
# ---
